Use options.accept and contentType when provided

viniciuskneves opened the pull request:
Related to: https://github.com/restify/clients/issues/127

We were facing issues when 'accept' header was something like: "application/vnd.trustvox-v2+json".

I think it should be possible to update both: 'accept' and 'contentType' headers as other ``StringClient`` (https://github.com/restify/clients/blob/master/lib/StringClient.js#L21)

I will also update documentation that is missing 'contentType' option.

viniciuskneves commented:
@souzawillian and @br-monteiro fyi

DonutEspresso commented:
Thanks for the help @viniciuskneves, massively appreciated!  🎉 

This looks great - would love to see a test here using a local test server that responds with a type that you are expecting to be parsed as JSON.

viniciuskneves commented:
@DonutEspresso I've just added some tests to ``JsonClient`` and all of the sudden the coverage dropped 60% hahaha

Any suggestions?

Btw, I've added the tests you asked for :stuck_out_tongue_closed_eyes:

DonutEspresso commented:
Sorry for the delay! You added a lot more tests than I expected :)  We certainly don't expect you to add tests for things that aren't already being tested, but thank you very much for that!

viniciuskneves commented:
@DonutEspresso thanks for your feedback!

Btw, I think the release of the new version is missing. Should I'd committed version update too?

DonutEspresso commented:
Hi - missed this comment as it was on a closed issue. We are cutting a release next week. Sorry for the delay!
